### PR TITLE
chore: update swift package tools [WPB-6090]

### PR DESCRIPTION
**Pull request by markusfassbender**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6090" title="WPB-6090" target="_blank">TaskWPB-6090</a>  [iOS] Migrate to Xcode 15.2
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Document how to update the swift package tools
- Update [Sourcery](https://github.com/krzysztofzablocki/Sourcery)
  - Add missing run phase to `WireDataModel`
- Update [SwiftLint](https://github.com/realm/SwiftLint)
  - Fix `redundant_void_return` rules, see [release notes](https://github.com/realm/SwiftLint/releases/tag/0.54.0).

#### How to Test

No testing required.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by agisilaos — COMMENTED** [review]
@markusfassbender Please bump the SwiftLint version in .swiftlint.yml file too.

File: scripts/run-sourcery.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ set -Eeuo pipefail
 
 REPO_ROOT=$(git rev-parse --show-toplevel)
 SCRIPTS_DIR="$REPO_ROOT/scripts"
-SOURCERY="$SCRIPTS_DIR/.build/artifacts/scripts/sourcery/sourcery.artifactbundle/sourcery/bin/sourcery/bin/sourcery"
+SOURCERY="$SCRIPTS_DIR/.build/artifacts/scripts/sourcery/sourcery/bin/sourcery"
```
Comment (netbe) [inline scripts/run-sourcery.sh:24]: why did that change?

Comment (markusfassbender) [reply scripts/run-sourcery.sh:24]: @caldrian helped me this this, do you know?

Comment (caldrian) [reply scripts/run-sourcery.sh:24]: The structure in the zip file changed upstream. I hope they'll keep the structure from now on.

File: wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+Participants.swift [hunk 2]
```diff
@@ -555,7 +555,7 @@ final class ConversationParticipantsTests: ZMConversationTestsBase {
 
     func testThatItRefetchesRolesIfNoRoles() {
 
-        syncMOC.performGroupedAndWait { _ -> Void in
```
Comment (caldrian) [inline wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+Participants.swift:558]: Nice, however, maybe these changes should have gone into a separate PR.

**Review by caldrian — APPROVED** [review]

**Review by netbe — APPROVED** [review]

File: wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+Participants.swift (hunk 2 above)
Comment (markusfassbender) [reply wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+Participants.swift:558]: The rule got rewritten with version 0.54.0 and thus lead to failure. Either I could change the swiftlint configuration or just solve the cases...

**Pull request merged**